src/typescript-worker/language-service-context.ts

```diff
@@ -0,0 +1,140 @@
+/**
+ * @license
+ * Copyright 2021 Google LLC
+ * SPDX-License-Identifier: BSD-3-Clause
+ */
+
+import ts from '../internal/typescript.js';
+import {ImportMapResolver} from './import-map-resolver.js';
+
+import type {CachingCdn} from './caching-cdn.js';
+
+const compilerOptions = {
+  target: ts.ScriptTarget.ES2017,
+  module: ts.ModuleKind.ESNext,
+  experimentalDecorators: true,
+  skipDefaultLibCheck: true,
+  skipLibCheck: true,
+  moduleResolution: ts.ModuleResolutionKind.NodeJs,
+  lib: ['dom', 'esnext'],
+};
+
+/**
+ * Compiles a project, returning a Map of compiled file contents. The map only
+ * contains context for files that are compiled. Other files are skipped.
+ *
+ * TODO (justinfagnani):  We could share a DocumentRegistry across
+ * multiple <playground-project> instances to save memory and type analysis of
+ * common lib files like lit-element, lib.d.ts and dom.d.ts.
+ */
+export class LanguageServiceContext {
+  public readonly compilerOptions = compilerOptions;
+  public readonly cdn: CachingCdn;
```

Comment:
It seems like `CachingCdn` and `ImportMapResolver` probably don't quite belong in this class, because they're not very closely tied to the TypeScript/LanguageService: [1] they're used by the `BareModuleBuilder` which isn't TypeScript/LanguageService-related and [2] none of the methods here actually need access to the `CachingCdn` or `ImportMapResolver`

How about factoring out one more class -- something like this?:

- Rename `project-cache.ts` to `worker-context.ts`
- Add a new class called `WorkerContext` inside `worker-context.ts`
- The new class has these fields:
  - `CachingCdn`
  - `ImportMapResolver`
  - `LanguageServiceContext`
- Rename `getLanguageServiceContext` to `getWorkerContext`.